src/components/Checkbox/__tests__/Checkbox.test.tsx

```diff
@@ -0,0 +1,191 @@
+    });
+
+    test('set given title to label', () => {
+        const title = 'Some title';
+
+        render(<Checkbox title={title} />);
+        const label = screen.getByTitle(title);
+
+        expect(label).toBeVisible();
+    });
+
+    test('show given content', () => {
+        const content = 'Some content';
+
+        render(<Checkbox content={content} />);
+        const text = screen.getByLabelText(content);
+
+        expect(text).toBeVisible();
+    });
+
+    test('show given children', () => {
+        const childrenText = 'Children content';
+
+        render(
+            <Checkbox>
+                <span>{childrenText}</span>
+            </Checkbox>,
+        );
+        const text = screen.getByLabelText(childrenText);
+
+        expect(text).toBeVisible();
+    });
+
+    test('add className', () => {
+        const className = 'my-class';
+
+        render(<Checkbox className={className} qa={qaId} />);
+        const component = screen.getByTestId(qaId);
+
+        expect(component).toHaveClass(className);
+    });
+
+    test('add style', () => {
+        const style = {color: 'red'};
+
+        render(<Checkbox style={style} qa={qaId} />);
+        const component = screen.getByTestId(qaId);
+
+        expect(component).toHaveStyle(style);
+    });
+
+    test('use defaultChecked attribute', () => {
+        render(<Checkbox defaultChecked={true} />);
+        const checkbox = screen.getByRole('checkbox');
+
+        expect(checkbox).toBeChecked();
+    });
+
+    test('set checked=true attribute for controlled component', () => {
+        render(<Checkbox checked={true} />);
+        const checkbox = screen.getByRole('checkbox');
+
+        expect(checkbox).toBeChecked();
+    });
+
+    test('set checked=false attribute for controlled component', () => {
+        render(<Checkbox checked={false} />);
+        const checkbox = screen.getByRole('checkbox');
+
+        expect(checkbox).not.toBeChecked();
+    });
+
+    test('set base control props', () => {
+        const id = 'my_id';
+        const name = 'my name';
+        const value = 'my value';
+
+        render(<Checkbox id={id} name={name} value={value} />);
+        const checkbox = screen.getByRole('checkbox');
+
+        expect(checkbox).toHaveAttribute('id', id);
+        expect(checkbox).toHaveAttribute('name', name);
+        expect(checkbox).toHaveAttribute('value', value);
+    });
+
+    test('use passed ref for component', () => {
+        const ref = React.createRef<HTMLLabelElement>();
+
+        render(<Checkbox ref={ref} qa={qaId} />);
+        const component = screen.getByTestId(qaId);
+
+        expect(ref.current).toBe(component);
+    });
+
+    test('toggle checked attribute on click', async () => {
+        const text = 'I agree';
+        const user = userEvent.setup();
+
+        render(<Checkbox content={text} />);
+        const checkbox = screen.getByRole('checkbox');
+        const caption = screen.getByLabelText(text);
```

Comment:
here `checkbox === caption`, `getByLabelText` returns an element associated with label.

Reply:
Oh, yes, thanks. I've missed that replacing [by this comment](https://github.com/yandex-cloud/uikit/pull/253#discussion_r925405158). Returned to `getByText`